src/app/tracing/graph/graph.service.ts

```diff
@@ -602,7 +602,7 @@ export class GraphService {
     }
 
     private updateHoverEdges(state: SharedGraphState, newData: GraphServiceData): void {
-        newData.hoverEdges = state.hoverDeliveries.map(delId => newData.delIdToEdgeDataMap[delId].id);
+        newData.hoverEdges = state.hoverDeliveries.map(delId => newData.delIdToEdgeDataMap[delId]?.id);
```

Comment:
I felt that instead of accessing keys on an object we are passed from another source and don't know for sure exist on the object, and afterwards filtering out the values where undefined was returned, it would be better to only access keys that exists on the object. 
This way instead of building code that might fail to access a key but would suppress the error, we make it so the code only accesses existing keys.

```suggestion
        newData.hoverEdges = state.hoverDeliveries
            .filter(delId => delId in newData.delIdToEdgeDataMap)
            .map(delId => newData.delIdToEdgeDataMap[delId].id);
```

Reply:
Ok.